grails-app/controllers/UserGroupController.groovy

```diff
@@ -42,6 +43,8 @@ class UserGroupController {
     def delete = {
         def userGroupInstance = UserGroup.get(params.id)
         if (userGroupInstance) {
+            def accessList = SecureObjectAccess.findByPrincipal(userGroupInstance)
+            accessList.each { it.delete(flush: true) }
```

Comment:
Controllers are not transactional by default. If one the deletes fails the operation will be half completed. This should either be marked `@Transactional` or (maybe better) moved to a service.